src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,165 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode;
+        if (isEmpty()) {
+            newNode = new Node<>(null, value, null);
+            tail = newNode;
```

Comment:
head is first element of the list. So if the size == 0 ----> head = newNode;
```suggestion
            head = newNode;
        } else {
            tail.next = newNode;
        }
        tail = newNode;
        size++;
```